### DISCUSSION TITLE
building pdrs and fars in CreateSession

### DISCUSSION THREAD

**Discussion by ranomine** [nextmn/.github#7] [Q&A] [answered]
Hi @louisroyer,

Thanks a lot for your nice code.
Would you have by chance a small example of building fars and pdrs MAPs to be used in CreateSession?

Many thanks

Roch.

**Comment by louisroyer** [comment, marked as answer]
To create a `FarMap`, you have to first create an array of [`CreateFar`](https://github.com/wmnsk/go-pfcp/blob/master/ie/create-far.go) IEs, and then you can call `NewFARMap(fars []*ie.IE) (farmap *FARMap, err error, cause uint8, offendingIE uint16)` 
```golang
	import "github.com/wmnsk/go-pfcp/ie"
	createFar := ie.NewCreateFAR(
		ie.NewFARID(1), // FAR ID is 1
		ie.NewApplyAction(0x01), // set the DROP flag
	),
	f := make([]*ie.IE, 0)
	f = append(f, createFar)
	farMap, _, _,_ := NewFARMap(f)
	
	// same (but not detailed here) for pdrMap
	// …
	a.CreateSession(pdrMap, farMap)


```
This is the same thing for PDRs with `NewPDRMap(pdrs []*ie.IE) (pdrmap *PDRMap, err error, cause uint8, offendingIE uint16)`, and `CreatePDR` IE (containing at least a PDR ID, a Precedence, a PDI and a FAR ID).

If your question was about what they should contain exactly, everything is detailed on TS 129.244.

**Comment by ranomine** [comment]
Perfect!

Merci beaucoup :)
